src/server/misc/cmds/cmds.go

```diff
@@ -0,0 +1,128 @@
+package cmds
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httputil"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/cmdutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/promutil"
+	"github.com/spf13/cobra"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+func Cmds() []*cobra.Command {
+	var commands []*cobra.Command
+
+	var d net.Dialer
+	setupControl(&d)
+
+	r := new(net.Resolver)
+	d.Resolver = r
+	r.Dial = func(ctx context.Context, network, address string) (_ net.Conn, retErr error) {
+		defer log.Span(ctx, "DialDNS", zap.String("network", network), zap.String("address", address))(log.Errorp(&retErr))
+		return d.DialContext(ctx, network, address) //nolint:wrapcheck
+	}
+
+	t := http.DefaultTransport.(*http.Transport).Clone()
+	t.DialContext = d.DialContext
+	t.TLSClientConfig.VerifyConnection = func(cs tls.ConnectionState) error {
+		fmt.Printf("tls: server name: %v\n", cs.ServerName)
+		fmt.Printf("tls: negotiated protocol: %v\n", cs.NegotiatedProtocol)
+		fmt.Println()
+		return nil
+	}
+	t.TLSClientConfig.InsecureSkipVerify = true
+	hc := &http.Client{
+		Transport: promutil.InstrumentRoundTripper("pachctl", t),
+	}
+
+	dnsLookup := &cobra.Command{
+		Use:   "{{alias}} <hostname>",
+		Short: "Do a DNS lookup on a hostname.",
+		Long:  "Do a DNS lookup on a hostname.",
+		Run: cmdutil.RunFixedArgs(1, func(args []string) error {
+			ctx := pctx.Background("")
+			var errs error
+			if result, err := r.LookupHost(ctx, args[0]); err != nil {
+				multierr.AppendInto(&errs, errors.Wrap(err, "lookup host"))
+			} else {
+				fmt.Printf("A: %v\n", result)
+			}
+			if result, err := r.LookupCNAME(ctx, args[0]); err != nil {
+				multierr.AppendInto(&errs, errors.Wrap(err, "lookup cname"))
+			} else {
+				fmt.Printf("CNAME: %v\n", result)
+			}
+			if result, err := r.LookupAddr(ctx, args[0]); err != nil {
+				multierr.AppendInto(&errs, errors.Wrap(err, "lookup reverse address"))
+			} else {
+				fmt.Printf("IP: %v\n", result)
+			}
+			if errs != nil {
+				fmt.Printf("\nDo not be alarmed by the below errors; some lookups are expected to fail.\n")
```

Comment:
I have no strong feelings about making stdout / stderr / the exit code clean.  This will be run during debugging sessions with customers; so only the sum total of the text matters.  Why did I add this warning then?  Dunno.  I think I should delete it.

Reply:
I cleaned this up.  We print the messages to stderr, one per line, and don't return them.